heron/healthmgr/src/java/com/twitter/heron/healthmgr/common/PackingPlanProvider.java

```diff
@@ -0,0 +1,108 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.healthmgr.common;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.logging.Logger;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+
+import com.microsoft.dhalion.events.EventHandler;
+import com.microsoft.dhalion.events.EventManager;
+
+import com.twitter.heron.healthmgr.common.HealthManagerEvents.TopologyUpdate;
+import com.twitter.heron.proto.system.PackingPlans;
+import com.twitter.heron.spi.packing.PackingPlan;
+import com.twitter.heron.spi.packing.PackingPlan.ContainerPlan;
+import com.twitter.heron.spi.packing.PackingPlan.InstancePlan;
+import com.twitter.heron.spi.packing.PackingPlanProtoDeserializer;
+import com.twitter.heron.spi.statemgr.SchedulerStateManagerAdaptor;
+
+import static com.twitter.heron.healthmgr.HealthManager.CONF_TOPOLOGY_NAME;
+
+/**
+ * A topology's packing plan may get updated after initial deployment. This provider is used to
+ * fetch the latest version from the state manager and provide to any dependent components.
+ */
+public class PackingPlanProvider implements Provider<PackingPlan>, EventHandler<TopologyUpdate> {
+  private static final Logger LOG = Logger.getLogger(PackingPlanProvider.class.getName());
+
+  private final SchedulerStateManagerAdaptor stateManagerAdaptor;
+  private final String topologyName;
+
+  private PackingPlan packingPlan;
+
+  @Inject
+  public PackingPlanProvider(SchedulerStateManagerAdaptor stateManagerAdaptor,
+                             EventManager eventManager,
+                             @Named(CONF_TOPOLOGY_NAME) String topologyName) {
+    this.stateManagerAdaptor = stateManagerAdaptor;
+    this.topologyName = topologyName;
+    eventManager.addEventListener(TopologyUpdate.class, this);
+  }
+
+  public String[] getBoltInstanceNames(String... boltComponents) {
+    HashSet<String> boltComponentNames = new HashSet<>();
+    for (String boltComponentName : boltComponents) {
+      boltComponentNames.add(boltComponentName);
+    }
+
+    PackingPlan packing = get();
+    ArrayList<String> boltInstanceNames = new ArrayList<>();
+    for (ContainerPlan containerPlan : packing.getContainers()) {
+      for (InstancePlan instancePlan : containerPlan.getInstances()) {
+        if (!boltComponentNames.contains(instancePlan.getComponentName())) {
+          continue;
+        }
+
+        String name = "container_" + containerPlan.getId()
```

Comment:
The instance id `container_%s_%s_%d` seems to be set here https://github.com/twitter/heron/blob/master/heron/executor/src/python/heron_executor.py?utf8=%E2%9C%93#L568 and then used by `MetricsManagerServer` to be sent to tracker. I did not find any other user of this string.